### ISSUE TITLE
[FTR] Filtering posts using categories

### ISSUE THREAD

**Issue by nikdhin** [austinhuang0131/barinsta#328]
Hey there, 

Thank you for developing Barinsta! In my opinion, I think a big feature that would be really useful (and good for reducing time spent on IG) is the ability to group accounts you follow into categories. That way the Feed would be able to display posts from accounts only in a specific category. 

**Is your feature request related to a problem? Please describe.**
Feature: Allowing a user to group accounts they follow into categories like Friends, Art, etc.
Use Case: Currently, if I wanna catch up on my friends' posts I still have to scroll a lot since I follow accounts related to my other interests. 

**Describe the solution you'd like**
Allow creation of custom groups/categories like "Friends", "Art", etc, as well as the ability to view posts from accounts only in 
a specific category.

Please let me know if you have any questions or concerns.

**Comment by austinhuang0131** [comment]
tldr: Such feature would not be scalable.

---

Before I explain, you must understand that:

1. Feed is 1 endpoint. It is paginated, which means there is a limit of how many posts I can fetch per request.
2. Each user is 1 endpoint. It is also paginated.

This gives us 2 options:

1. Request feed and filter out desired users.
2. Request each desired user. => See #24

The first method will have these problems:

1. Since there is no guarantee that each feed page contains posts from your desired users (especially if you have many followed users - hence not scalable), you might need to fetch several pages before getting results. Fetching them too fast may cause ratelimiting (which can lead to a ban). Fetching them too slow will render this feature unnecessary (and bad for UX).
2. Feed is not chronological. Feed is paginated, and while I can sort posts within a page, I can't sort pages.

No, I can't just use a RSS bridge either: That'll just move the ratelimit problem onto me and will eventually backfire. Again, I certainly would _like_ such a feature, but I also need to ensure user safety (given how strict Instagram enforces against abnormal behaviour) and UX

**Comment by nikdhin** [comment]
Gotcha, that makes sense. Thank you for going into the details! That was really insightful
